core/src/main/java/tech/pegasys/eth2signer/core/multikey/metadata/ArtifactSignerFactory.java

```diff
@@ -135,6 +142,30 @@ private ArtifactSigner createHashicorpArtifact(final HashicorpSigningMetadata me
     }
   }
 
+  private ArtifactSigner createAzureArtifact(final AzureSigningMetadata metadata) {
+
+    try {
+      final AzureKeyVault azureVault =
+          new AzureKeyVault(
+              metadata.getClientId(),
+              metadata.getClientSecret(),
+              metadata.getTenantId(),
+              metadata.getVaultName());
+
+      final Optional<String> secret = azureVault.fetchSecret(metadata.getSecretName());
+      if (secret.isEmpty()) {
+        throw new SigningMetadataException(
+            "secret '" + metadata.getSecretName() + "' doesn't exist");
+      }
+      final BLSKeyPair keyPair =
```

Comment:
do we want to wrap this up in "caching" signer - such that the secret key doesn't stay in memory - but rather fetched everytime? 
Or is that implicit in our whole design? I.e. we ALWAYS go via this class to get a signer? (and thus reconstruct it everytime)